File: Data Structures & Algorithms/longest-common-prefix/submission-3.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        Dictionary<string, int> dict = new Dictionary<string, int>();
        String longest = "";
        for (int i = 0; i < strs.Length; i++) {
            for (int j = 0; j <= strs[i].Length; j++) {
                String temp = strs[i].Substring(0, j);
                if (dict.ContainsKey(temp)) {
                    dict[temp] += 1;
                    if (dict[temp] == strs.Length) {
                        longest = temp;
                    }
                }
                else {
                    dict[temp] = 1;
                    if (dict[temp] == strs.Length) {
                        longest = temp;
                    }
                }
            }
        }
        return longest;
    }
}

File: Data Structures & Algorithms/non-cyclical-number/submission-3.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = {}
        temp = n

        while (temp != 1):
            curT = 0
            while (temp > 0):
                curT += (temp % 10) ** 2
                temp //= 10
            if curT in seen:
                return False
            seen[curT] = True
            temp = curT
        return True
